src/App.js

```diff
@@ -13,25 +15,76 @@ class App extends React.Component {
     super(props);
 
     this.state = {
-      phones: getAll(),
+      phones: [],
       selectedPhone: null,
       basketItems: [],
-    };
+    }
   }
 
+/*  componentWillMount() {
+    localStorage.getItem('basketItems') && this.setState({
+      basketItems: localStorage.getItem('basketItems')
+    });
+  }*/
+
+  componentWillUpdate = (nextProps, nextState) => {
+    console.log(this.state.basketItems);
+    localStorage.getItem('basketItems') && localStorage.setItem('basketItems', this.state.basketItems)
+    console.log(this.state.basketItems);
+    localStorage.setItem('basketItemsDate', Date.now());
+  }
+
+  updateData = (value) => {
+    this.setState({
+      basketItems: [...this.state.basketItems , value]})
+  }
+
+  deleteData = (value) => {
+    const array=[...this.state.basketItems];
+    array.splice(array.indexOf(value), 1);
+    this.setState({basketItems: array})
+  }
+  
+  countSameEls = (value) => {
+    let result = 0;
+    this.state.basketItems.map(item => (item === value) ? (result += 1) : null)
```

Comment:
Don't use `map` if you don't expect to return new value from array.
```suggestion
    this.state.basketItems.forEach(item => {
      if (item === value) {
        result += 1
      }
    })
```